module/min/minwgpu/src/context.rs

```diff
@@ -0,0 +1,503 @@
+//! This module provides a fluent, type-state builder for setting up a `wgpu` context,
+//! which includes the `Instance`, `Adapter`, `Device`, and `Queue`. This pattern guides
+//! the user through the required steps of `wgpu` initialization in the correct order.
+
+use mingl::mod_interface;
+
+mod private
+{
+  use core::marker::PhantomData;
+
+  /// A container for the core `wgpu` components, representing a complete graphics context.
+  ///
+  /// An instance of `Context` holds everything needed to start creating resources and rendering.
+  pub struct Context
+  {
+    pub( super ) instance : wgpu::Instance,
+    pub( super ) adapter : wgpu::Adapter,
+    pub( super ) device : wgpu::Device,
+    pub( super ) queue : wgpu::Queue,
+  }
+
+  impl Context
+  {
+    /// Creates a new `ContextBuilder` to start the setup process.
+    ///
+    /// This is the entry point for the fluent builder pattern.
+    #[ inline ]
+    #[ must_use ]
+    pub fn builder() -> ContextBuilder< 'static, 'static, 'static, 'static, InstanceBuilder >
+    {
+      ContextBuilder
+      {
+        _state : PhantomData,
+        instance_descriptor : wgpu::InstanceDescriptor::default(),
+        request_adapter_options : wgpu::RequestAdapterOptionsBase::default(),
+        device_descriptor : wgpu::wgt::DeviceDescriptor::default(),
+        instance : None,
+        adapter : None,
+        adapter_selector : None
+      }
+    }
+
+    /// Returns a reference to the `wgpu::Instance`.
+    #[ inline ]
+    #[ must_use ]
+    pub fn instance( &self ) -> &wgpu::Instance
+    {
+      &self.instance
+    }
```

Comment:
For getters you should write get_instance, get_instance_mut
So it is visible that you return immutable reference